tower_cli/cli/transfer/send.py

```diff
@@ -0,0 +1,1200 @@
+                            asset_changed = True
+                            self.log_change("Updated asset")
+                        except TowerCLIError as e:
+                            self.log_error("Failed to update {} {} : {}".format(asset_type, asset_name, e))
+                            continue
+                    else:
+                        self.log_ok("Asset up to date")
+
+                # If there are relations, import them
+                if relations is not None:
+                    for a_relation in relations:
+                        if a_relation == 'survey_spec':
+                            survey = tower_cli.get_resource(asset_type).survey(existing_object['id'])
+                            if survey != relations[a_relation]:
+                                self.log_change("Updating survey")
+                                resource.modify(pk=existing_object['id'], survey_spec=relations[a_relation])
+                            else:
+                                self.log_ok("Survey up to date")
+                        elif a_relation == 'workflow_nodes':
+                            # In can_object_post, we deep copy the initial set of nodes and add it to the
+                            # end of the array This enables us to compare the exported nodes from the target
+                            # server to the unresolved nodes in the original import request
+                            # This has to be done because the can_object_post method resolves all of the dependencies
+                            existing_workflow_nodes = common.extract_workflow_nodes(existing_object)
+                            new_workflow_nodes_unresolved = relations[a_relation].pop()
+
+                            if not self.are_workflow_nodes_the_same(
+                                    existing_workflow_nodes, new_workflow_nodes_unresolved
+                            ):
+                                self.import_workflow_nodes(existing_object, relations[a_relation])
+                            else:
+                                self.log_ok("Workflow nodes up to date")
+                        elif a_relation == 'host' or a_relation == 'inventory_source':
+                            self.import_inventory_relations(existing_object, relations[a_relation], a_relation)
+                        elif a_relation == 'group':
+                            self.import_inventory_groups(existing_object, relations[a_relation])
+                        elif a_relation in common.NOTIFICATION_TYPES:
+                            self.import_notification_relations(existing_object, relations[a_relation], a_relation)
+
+                # Checking for post update actions on the different objects
+                if asset_changed:
+                    if asset_type == 'project':
+                        try:
+                            resource.update(existing_object['id'], wait=True)
```

Comment:
I'm not sure why this is strictly necessary. Projects will automatically update when created, and if copying the specified branch/hash of the original should be ideal. I understand that you probably need to wait until it updated before you can create related resources. I think `wait` is also a kwarg for project create.

Reply:
As you said, the update has to happen to make sure Tower has whatever playbooks are in the project so that we can build any subsequent job_templates. The wait in kwargs would work if we were building a new project but what if the project is pre-existing and the import process is changing the SCM branch? From a quick look at the models, I don't think an update happens on a project write. Can you confirm that?

Reply:
We had an AWX issue somewhere to automatically trigger an update if the branch was changed... I would love for AWX to fix its current behavior instead of this, but I see where you're coming from. As long as no-op runs don't bog themselves down with project updates I see how this makes sense now.